File: OpenEvent.Web/Models/Event/Event.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;
using OpenEvent.Web.Models.Analytic;
using OpenEvent.Web.Models.Category;

namespace OpenEvent.Web.Models.Event
{
    /// <summary>
    /// Base event model with all data.
    /// </summary>
    public class Event
    {
        /// <summary>
        /// Unique id
        /// </summary>
        public Guid Id { get; set; }

        /// <summary>
        /// Event name
        /// </summary>
        public string Name { get; set; }

        /// <summary>
        /// Markdown description parsed to sting
        /// </summary>
        public string Description { get; set; }

        /// <summary>
        /// Event thumbnail
        /// </summary>
        public Image Thumbnail { get; set; }

        /// <summary>
        /// Event image gallery
        /// </summary>
        public List<Image> Images { get; set; }

        /// <summary>
        /// Social media links
        /// </summary>
        public List<SocialLink> SocialLinks { get; set; }

        /// <summary>
        /// Date time of creation
        /// </summary>
        public DateTime Created { get; set; }

        /// <summary>
        /// Starting time
        /// </summary>
        public DateTime StartLocal { get; set; }

        /// <summary>
        /// Universal starting time
        /// </summary>
        public DateTime StartUTC { get; set; }

        /// <summary>
        /// Ending time
        /// </summary>
        public DateTime EndLocal { get; set; }

        /// <summary>
        /// Universal ending time 
        /// </summary>
        public DateTime EndUTC { get; set; }

        /// <summary>
        /// base price of ticket saved as long, eg: £10.50 as 1050
        /// </summary>
        public long Price { get; set; }

        /// <summary>
        /// User who is hosting the event
        /// </summary>
        [JsonIgnore]
        public User.User Host { get; set; }

        /// <summary>
        /// Address if the event has one
        /// </summary>
        public Address.Address Address { get; set; }

        /// <summary>
        /// If the event is online
        /// </summary>
        public bool IsOnline { get; set; }

        /// <summary>
        /// List of all tickets associated with the event
        /// </summary>
        public List<Ticket.Ticket> Tickets { get; set; }

        /// <summary>
        /// Number of tickets not sold yet
        /// </summary>
        public int TicketsLeft { get; set; }

        /// <summary>
        /// List of event categories
        /// </summary>
        public List<EventCategory> EventCategories { get; set; }

        /// <summary>
        /// If the event has been canceled
        /// </summary>
        public bool isCanceled { get; set; }

        /// <summary>
        /// List of page views
        /// </summary>
        [JsonIgnore]
        public List<PageViewEvent> PageViewEvents { get; set; }

        /// <summary>
        /// List of ticket verifications
        /// </summary>
        [JsonIgnore]
        public List<TicketVerificationEvent> VerificationEvents { get; set; }

        /// <summary>
        /// List of all transactions associated with the event
        /// </summary>
        public List<Transaction.Transaction> Transactions { get; set; }

        /// <summary>
        /// List of event's promos
        /// </summary>
        public List<Promo.Promo> Promos { get; set; }

        /// <summary>
        /// If the event has finished
        /// </summary>
        public bool Finished { get; set; }
    }
}

File: OpenEvent.Web/ClientApp/src/app/_testData/Event.ts
import {EventHostModel} from "../_models/Event";
import {SocialMedia} from "../_models/SocialMedia";
import {Address} from "../_models/Address";

export const FakeEventHostModel: EventHostModel = {
  Promos: [],
  Transactions: [],
  Address: {AddressLine1: "", AddressLine2: "", City: "", CountryCode: "", CountryName: "", PostalCode: ""},
  Categories: [
    {Name: "Music"},
    {Name: "Performing"}
  ],
  Description: "this is a host event model",
  EndLocal: new Date(0),
  EndUTC: new Date(0),
  Id: "1",
  Images: [
    {Label: "test image", Source: "IMAGE"}
  ],
  IsOnline: false,
  Name: "tests host event",
  Price: 1010,
  SocialLinks: [
    {SocialMedia: SocialMedia.Site, Link: "http://www.example.com"},
    {SocialMedia: SocialMedia.Instagram, Link: "http://www.example.com"},
    {SocialMedia: SocialMedia.Twitter, Link: "http://www.example.com"},
    {SocialMedia: SocialMedia.Facebook, Link: "http://www.example.com"},
    {SocialMedia: SocialMedia.Reddit, Link: "http://www.example.com"}
  ],
  StartLocal: new Date(0),
  StartUTC: new Date(0),
  Thumbnail: {Label: "thumbnail", Source: "THUMBNAIL"},
  Tickets: [],
  TicketsLeft: 0
}

export const FakeAddress: Address = {
  AddressLine1: "Goldsmiths, University of London",
  AddressLine2: "",
  City: "London",
  CountryCode: "GB",
  CountryName: "United Kingdom",
  PostalCode: "SE14 6NW",
  Lat: 1,
  Lon: 1
}
